sqlglot/dataframe/sql/dataframe.py

```diff
@@ -0,0 +1,730 @@
+        output_expressions = []
+        replacement_mapping = {}
+        for expression_type, select_expression in select_expressions:
+            select_expression = select_expression.transform(replace_id_value, replacement_mapping)
+            if optimize:
+                select_expression = optimize_func(select_expression)
+            select_expression = df._replace_cte_names_with_hashes(select_expression)
+
+            # temporary hack to remove extra CTE that was replaced with table until removing unused CTEs is added
+            # to optimizer
+            filtered_ctes = [
+                cte
+                for cte in select_expression.ctes
+                if cte.alias_or_name not in {x.alias_or_name for x in replacement_mapping.values()}
+            ]
+            with_expr = exp.With(expressions=filtered_ctes) if filtered_ctes else None
+            select_expression.set("with", with_expr)
+
+            if expression_type == exp.Cache:
+                cache_table_name = df._create_hash_from_expression(select_expression)
+                cache_table = exp.to_table(cache_table_name)
+                original_alias_name = select_expression.args["cte_alias_name"]
+                replacement_mapping[exp.to_identifier(original_alias_name)] = exp.to_identifier(cache_table_name)
+                sqlglot.schema.add_table(cache_table_name, select_expression.named_selects)
+                cache_storage_level = select_expression.args["cache_storage_level"]
+                options = [
+                    exp.Literal(this="storageLevel", is_string=True),
+                    exp.Literal(this=cache_storage_level, is_string=True),
+                ]
+                expression = exp.Cache(this=cache_table, expression=select_expression, lazy=True, options=options)
+                # We will drop the "view" if it exists before running the cache table
+                output_expressions.append(exp.Drop(this=cache_table, exists=True, kind="VIEW"))
+            elif expression_type == exp.Create:
+                expression = df.output_expression_container.copy()
+                expression.set("expression", select_expression)
+            elif expression_type == exp.Insert:
+                expression = df.output_expression_container.copy()
+                select_without_ctes = select_expression.copy()
+                select_without_ctes.set("with", None)
+                expression.set("expression", select_without_ctes)
+                if select_expression.ctes:
+                    expression.set("with", exp.With(expressions=select_expression.ctes))
+            elif expression_type == exp.Select:
+                expression = select_expression
+            else:
+                raise ValueError(f"Invalid expression type: {expression_type}")
+            output_expressions.append(expression)
+
+        return [expression.sql(**{"dialect": dialect, "pretty": True, **kwargs}) for expression in output_expressions]
```

Comment:
pretty default

Reply:
Is the issue that it is pretty by default? I think users of the DataFrame API will want pretty by default. I'm not expecting many use cases where individuals will want the minimal SQL returns for performance reasons.

Reply:
yep, consistency

Reply:
so let's get rid of the pretty default because everywhere else pretty is not default